### ISSUE TITLE
Server access log garbled

### ISSUE THREAD

**Issue by jinyu121** [v2ray/v2ray-core#572]
1)   What version of V2Ray are you using (If you deploy different version on server and client, please explicitly point out)?

latest

3)  What did you see?

Log garbled

I am reviewing the access log on the server, and notice some(just these 3 items) items like this:

```
accepted tcp:ä©^?<8e><87>^EW³;®F<9b>wàúZ<84>^^kó^LXP>©è
<86>Ë^R£½+T<86>MD¯;^MðQë\$£PSÖèÕI<86>^W¯µ¢Õ<83>¼Û!^^¿þ<9b>ía;õÂ:6994
accepted tcp:<97>^Lò.Amàâ}Å{íB <94>A£ÿmA¡Û<%}^^Ä^XQHÈæQÏ^EÜUÇ<8e>¤^P4^PW<9c>súØh(^YÈ«zQá²\çU*É0^Y^Q:45755
```

n) Other

Server configuration:

```
"log": {
        "access": "/var/log/v2ray/access.log",        // I am reviewing this log file
        "error": "/var/log/v2ray/error.log",
        "loglevel": "warning"
    }
```

I am not sure which inbound/outbound caused this problem, since I have VMess+dynamic port and several ShadowSocks inbounds. The outbound protocol is "freedom".

**Comment by DarienRaymond** [comment]
Did you visit any domain containing non-ASCII characters? it looks the file encoding is not correct.

**Comment by jinyu121** [comment]
Well, I opened the log file using both vim and cat, and saw the same problem. 

When VR create log file, what encoding is used?

I am sure I haven't visit any websites with non-ASCII chars in domains.

**Comment by xiaokangwang** [comment]
Is that possible that some hostile traffic passed vmess authorization?

**Comment by xiaokangwang** [comment]
Even domains with Unicode name should transmit to V2Ray with a ascii form of it. And it is really unlikely that any valid domain name will have no dot at all(Even if encoding is wrong, some ascii characters should be preserved).

**Comment by DarienRaymond** [comment]
Not sure. I think it is an internal bug, or caused by socks doesn't handle long domain properly.

**Comment by DarienRaymond** [comment]
Should be fixed in 2.42.
